### ISSUE TITLE
CA2235 shouldn't fire for fields implementing ISerializable

### ISSUE THREAD

**Issue by drewnoakes** [dotnet/roslyn-analyzers#1970]
#### Analyzer package

[Microsoft.NetFramework.Analyzers](https://www.nuget.org/packages/Microsoft.NetFramework.Analyzers)

#### Package Version

2.6.3-beta1.18614.2

#### Diagnostic ID

[CA2235 Mark all non-serializable fields](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2235-mark-all-non-serializable-fields)

#### Repro steps

```c#
internal sealed class NonSerializable
{
    public NonSerializable(string value) => Value = value;
    public string Value { get; }
}

[Serializable]
internal sealed class CustomSerializingType : ISerializable
{
    private readonly NonSerializable _nonSerializable; // CA2235 dislikes this field

    public CustomSerializingType(SerializationInfo info, StreamingContext context)
    {
        _nonSerializable = new NonSerializable(info.GetString("KEY"));
    }

    public void GetObjectData(SerializationInfo info, StreamingContext context)
    {
        info.AddValue("KEY", _nonSerializable.Value);
    }
}
```

#### Expected behavior

No diagnostic.

#### Actual behavior

Diagnostic.

**Comment by drewnoakes** [comment]
@mavasani there are other kinds of custom serialization:

https://docs.microsoft.com/en-us/dotnet/standard/serialization/custom-serialization#running-custom-methods-during-and-after-serialization

Do you think these need covering too? If so, I can open other issue(s).

**Comment by mavasani** [comment]
Sounds useful to me!

**Comment by drewnoakes** [comment]
@mavasani I was mistaken. Those attributed methods don't influence serialization of fields. They're just lifecycle callbacks, without access to a `SerializationInfo`. I don't see any action to take here, so will leave it.
